R/expand_model_out_grid.R

```diff
@@ -303,7 +319,17 @@ null_taskids_to_na <- function(model_task) {
 # Adds example sample ids to the output type id column which are unique
 # across multiple modeling task groups. Only apply to v3 and above sample output
 # type configurations.
-add_sample_idx <- function(x, round_config, config_tid) {
+add_sample_idx <- function(x, round_config, config_tid, compound_taskid_set = NULL) {
+  if (!is.null(compound_taskid_set) && length(compound_taskid_set) != length(x)) {
+    cli::cli_abort(
+      c("x" = "The length of {.var compound_taskid_set}
+      ({.val {length(compound_taskid_set)}})
+      must match the number of modeling tasks ({.val {length(x)}})
+        in the round."),
```

Comment:
Similar question to the one I asked on the other PR: suppose a hub defines a config file with one task group for flu and another for covid, with values of the task id variable `pathogen` being different between the groups. one could imagine wanting to override the compound taskid set to not include `pathogen`, right?

Reply:
I'm not 100% sure I follow the question but as mentioned before, submitted data that refer to different modeling tasks are processed separately and can have their own `compound_taskid_set`. That's also why when we want to override them we provide a list with a character vector of compound task IDs for each modeling task.